openedx/core/djangoapps/courseware_api/views.py

```diff
@@ -611,6 +613,10 @@ def get(self, request, usage_key_string, *args, **kwargs):  # lint-amnesty, pyli
             disable_staff_debug_info=True,
             will_recheck_access=True)
 
+        if not hasattr(sequence, 'get_metadata'):
```

Comment:
Is this the problem that we were looking at before the break? Is there anything useful we could log here to determine anything about who or what is making the bad call?

Reply:
This is one of the problems we were looking at before the break, yeah. The one that was surfacing as an AttributeError and we didn't know why it was happening more.

We could log this call (but that would probably be best done in the other MFE PR - so that we could send on the referrer of the parent page, not the referrer of this API call). But once I realized that this URL pattern (course+unit, instead of course+sequence+unit) is expected and supported, I started caring less about finding out who is calling it. Though it still strikes me as an unnecessary thing to allow.

If we ever wanted to stop supporting that URL structure, maybe we could add some logging then, to notify callers. But now that it's out there and supported, it's kind of a public API that we should keep doing _something_ reasonable for. So I'm content to let this sit like this for now and not bother finding out who is doing it.